src/Entity/User.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Validator\Constraints as Assert;
+use Webauthn\PublicKeyCredentialSource;
+use Webauthn\PublicKeyCredentialUserEntity;
+
+/**
+ * @ORM\Table(name="users")
+ * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
+ */
+class User extends PublicKeyCredentialUserEntity implements UserInterface
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="string", length=255)
+     */
+    protected $id;
+
+    /**
+     * @Assert\Length(max = 100)
+     */
+    protected $name;
+
+    /**
+     * @Assert\Length(max = 100)
+     */
+    protected $displayName;
+
+    /**
+     * @var PublicKeyCredentialSource[]
+     * @ORM\ManyToMany(targetEntity="App\Entity\PublicKeyCredentialSource")
+     * @ORM\JoinTable(name="users_user_handles",
+     *      joinColumns={@ORM\JoinColumn(name="user_id", referencedColumnName="id")},
+     *      inverseJoinColumns={@ORM\JoinColumn(name="user_handle", referencedColumnName="id", unique=true)}
+     *      )
+     */
+    protected $publicKeyCredentialSources;
+
+    public function __construct(string $id, string $name, string $displayName)
+    {
+        parent::__construct($name, $id, $displayName);
+        $this->publicKeyCredentialSources = new ArrayCollection();
+    }
+
+    /**
+     * Webauthn project does not care about roles of any user.
+     *
+     * @return array
+     */
+    public function getRoles(): array
```

Comment:
Why are these methods created?

Reply:
Because https://github.com/web-auth/webauthn-framework base class requires these methods to exists. This is only for Symfony firewall support, we don't use this functionality.